.github/workflows/ci.yml

```diff
@@ -65,25 +95,96 @@ jobs:
       - run: corepack enable
       - run: yarn install
 
-      - name: build test images
+      - name: build test images ${{ matrix.platform }} == ${{ env.DEFAULT_PLATFORM }}
         run: |
           docker info
-          node_modules/.bin/synthetic-chain build
+          node_modules/.bin/synthetic-chain build ${{ matrix.platform == env.DEFAULT_PLATFORM && ' ' || '--dry' }}
       - name: run test images
+        if: ${{ matrix.platform == env.DEFAULT_PLATFORM }}
         run: node_modules/.bin/synthetic-chain test
 
-      # XXX this should be instant because all the stages were already built in the steps above
-      # but it's re-building the last stage. This is deemed good enough for now.
-      # see https://github.com/moby/moby/issues/34715
-      - name: Build and push complete image
+      - name: Compute tags
```

Comment:
Nit, can we move "compute tags" earlier in the job / before the build step so we can easily check the tags before the job completes?